src/main/java/org/jitsi/jigasi/transcription/OracleTranscriptionService.java

```diff
@@ -0,0 +1,353 @@
+     */
+    private final String websocketUrlConfig;
+
+
+    /**
+     * Create a TranscriptionService which will send audio to the OCI service
+     * platform to get a transcription
+     */
+    public OracleTranscriptionService()
+    {
+        websocketUrlConfig = JigasiBundleActivator.getConfigurationService()
+                .getString(WEBSOCKET_URL, DEFAULT_WEBSOCKET_URL);
+        compartmentId = JigasiBundleActivator.getConfigurationService()
+                .getString(COMPARTMENT_ID, "");
+    }
+
+    public boolean supportsLanguageRouting()
+    {
+        return false;
+    }
+
+
+    @Override
+    public boolean supportsStreamRecognition()
+    {
+        return true;
+    }
+
+    @Override
+    public boolean supportsFragmentTranscription()
+    {
+        return false;
+    }
+
+    @Override
+    public void sendSingleRequest(final TranscriptionRequest request,
+                                  final Consumer<TranscriptionResult> resultConsumer)
+    {
+        logger.warn("The OCI transcription service does not support single requests.");
+    }
+
+    @Override
+    public StreamingRecognitionSession initStreamingSession(
+            Participant participant)
+    {
+        return new OracleStreamingSession(websocketUrlConfig);
+    }
+
+    @Override
+    public boolean isConfiguredProperly()
+    {
+        return true;
+    }
+
+    public class OracleStreamingSession implements StreamingRecognitionSession, OracleRealtimeClientListener
+    {
+        private OracleRealtimeClient client;
+
+        private UUID uuid;
+
+        private boolean sessionEnding = false;
+
+        private final String websocketUrl;
+
+
+        /**
+         * List of TranscriptionListeners which will be notified when a
+         * result comes in
+         */
+        private final List<TranscriptionListener> listeners = new ArrayList<>();
+
+        private Instant sessionStart;
+
+
+        public OracleStreamingSession(String websocketUrl)
+        {
+            this.websocketUrl = websocketUrl;
+            uuid = Generators.timeBasedReorderedGenerator().generate();
+            try
+            {
+                client = new OracleRealtimeClient(
+                        this,
+                        authProvider,
+                        compartmentId);
+            }
+            catch (Exception e)
+            {
+                logger.error("Error while creating OCI client", e);
+            }
+        }
+
+        private void connect()
+        {
+            if (client.isConnected())
+            {
+                return;
+            }
+
+            final RealtimeParameters realtimeClientParameters = RealtimeParameters.builder()
+                    .isAckEnabled(false)
+                    .finalSilenceThresholdInMs(1000)
+                    .build();
+            try
+            {
+                client.open(websocketUrl, 443, realtimeClientParameters);
```

Comment:
DEFAULT_WEBSOCKET_URL is a full URL with port 8000, but here 443 is hardcoded?

Reply:
oracle will always be reachable on port 443 as they use a wss connection